### PR TITLE
794 issue templates

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://roche-gsk.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://roche-gsk.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://roche-gsk.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::documents()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://roche-gsk.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/02_feature_request.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/02_feature_request.yml:23]: Typo: "succint" instead of "succinct"

File: .github/ISSUE_TEMPLATE/02_feature_request.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/02_feature_request.yml:43]: Typo: I assume it should be "Tell us!".

File: .github/ISSUE_TEMPLATE/03_docs_update.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/03_docs_update.yml:30]: Should we add an "Other" category, e.g., for issues on other pages like "Changelog", "Home", or "Reference"?

File: .github/ISSUE_TEMPLATE/04_user_support.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/04_user_support.yml:7]: Should we mention that for bugs, enhancements, and documentation updates the corresponding issue types should be used?

File: .github/ISSUE_TEMPLATE/03_docs_update.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/03_docs_update.yml:36]: Should it be "Screenshots" to be consistent?

File: .github/ISSUE_TEMPLATE/04_user_support.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/04_user_support.yml:10]: I would remove the thread part of the link, i.e., use `https://app.slack.com/client/T028PB489D3/C02M8KN8269`

File: .github/ISSUE_TEMPLATE/05_onboard.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/05_onboard.yml:2]: I assume this issue should be created by Core members only.

Maybe we should mention this.

File: .github/ISSUE_TEMPLATE/02_feature_request.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/02_feature_request.yml:32]: What about the input?

Just an example output may be hard to understand without the corresponding input.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/02_feature_request.yml
Comment (bms63) [reply .github/ISSUE_TEMPLATE/02_feature_request.yml]: Fixed

File: .github/ISSUE_TEMPLATE/02_feature_request.yml
Comment (bms63) [reply .github/ISSUE_TEMPLATE/02_feature_request.yml]: So this doesn't show up in the template when it renders, but I fixed it.  You can see how they render by clicking on the files 
image

File: .github/ISSUE_TEMPLATE/03_docs_update.yml
Comment (bms63) [reply .github/ISSUE_TEMPLATE/03_docs_update.yml:30]: I added all those as categories.

File: .github/ISSUE_TEMPLATE/03_docs_update.yml
Comment (bms63) [reply .github/ISSUE_TEMPLATE/03_docs_update.yml:36]: Updated to Screenshot

File: .github/ISSUE_TEMPLATE/05_onboard.yml
Comment (bms63) [reply .github/ISSUE_TEMPLATE/05_onboard.yml:2]: I put message in it

File: .github/ISSUE_TEMPLATE/02_feature_request.yml
Comment (bms63) [reply .github/ISSUE_TEMPLATE/02_feature_request.yml:41]: Yes good point!

File: .github/ISSUE_TEMPLATE/04_user_support.yml
Comment (bms63) [reply .github/ISSUE_TEMPLATE/04_user_support.yml:10]: removed

**Comment by bms63** [comment]
Hi all,

@thomas-neitmann @rossfarrugia @koegerr @duniek.  Since we deal with the bulk of the "issues" in admiral, I thought you might want to take a look at these templates before they are merged into `devel`.  Stefan has given feedback on the PR.  Just wanted to see if any final additions are needed.  Can always update later, but we will be stuck with these until the next release.

I discovered that you can go directly to each of the five files in the `.github `folder to get the mock up of the template on Github.  Fun stuff!

https://github.com/pharmaverse/admiral/blob/794_issue_templates/.github/ISSUE_TEMPLATE/01_bug_report.yml

image

File: .github/ISSUE_TEMPLATE/04_user_support.yml
Comment (bundfussr) [inline .github/ISSUE_TEMPLATE/04_user_support.yml:2]: I think it shouldn't be a question mark at the end.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Comment by rossfarrugia** [comment]
@bms63 this is going to be a useful addition - thanks for all the efforts! here's some comments for consideration:

- for reproducible example sections can we add a note like _"REMINDER: no patient level data or company sensitive information should be shared via this open public issue"_. we have a similar note pinned to our Slack channel and i'm always keen to remind folks
- for feature request under feature idea, do we need the "(Succinct!)"? I read through a lot of external testing survey feedback comments where i was actually wishing the opposite and a bit more detail would have helped understand their needs. Love the idea you encourage them to give some pseudo code here, as hopefully that prompts them to contribute
- for docs update the category dropdown list is quite overwhelming - could simplify to something like: Function Documentation, User Guides, Developer Guides, Other. I'm sure that would be enough for us along with the details they add
- for user support on the top sentence about Slack you could add "... _- as for admiral we encourage support can come from anyone in our user community, not only our core developers_"
- i really like the onboarding page - good call on this one as will be a really useful checklist for future! for the non-core i don't think they need to be added to admiral_dev slack though - i thought we agreed that's only for the core? similarly i don't think they'd need access to our box folder - that's for the core team, and everything they need should be available to them in the Developer Guide site vignettes. final comment: use "GitHub" casing - Michael corrected me on that one for the pharmaverse site recently

great job!

**Review by bundfussr — APPROVED** [review]

**Comment by bms63** [comment]
@rossfarrugia I implemented your feedback on the templates.  @thomas-neitmann @duniek  any other things you want to see added in or updated/wordsmithed?

Thanks for the quick review @bundfussr!!

**Comment by bms63** [comment]
@thomas-neitmann or @bundfussr how do i do the check_templates  part?  Want to get this in unless others have opinions on the templates?

**Comment by rossfarrugia** [comment]
@bms63 there's no impact of this change to any ADaM templates so that can be ignored. i do think the check works as automated yet anyway, i've been ignoring it for now. i think you're fine to merge this one

**Comment by bundfussr** [comment]
check_templates should run when the PR is approved.

**Comment by thomas-neitmann** [comment]
You'll have to fix the merge conflict first.

**Comment by bms63** [comment]
@thomas-neitmann merge conflict fixed

File: .github/ISSUE_TEMPLATE/01_bug_report.yml
Comment (thomas-neitmann) [inline .github/ISSUE_TEMPLATE/01_bug_report.yml:17]: Do we really need this? You can only submit an issue with a GitHub account so we already have a line of communication.

File: .github/ISSUE_TEMPLATE/04_user_support.yml
Comment (thomas-neitmann) [inline .github/ISSUE_TEMPLATE/04_user_support.yml:3]: I do not think GitHub is the right place to ask for user support. In fact many repos explicitly state that you should not open issues asking how to do certain things. I feel like our Slack channel and (going forward) StackOverflow would be the right place to ask.

File: .github/ISSUE_TEMPLATE/05_onboard.yml
Comment (thomas-neitmann) [inline .github/ISSUE_TEMPLATE/05_onboard.yml:50]: No need for all these blank lines.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Comment by bms63** [comment]
@thomas-neitmann I removed the user support and email contact.  

@rossfarrugia I threw in the admiral 3 month community meeting into onboarding for both core and non-core.  Some gsk folks on the core team were unawares.  

For non-core members, should we also provide an invite for Question/Comments meeting if folks want to get clarification on their issue.  I put that in there for now.  Can take out.

**Comment by rossfarrugia** [comment]
@bms63 thanks - let me know if ever anyone from GSK needs to be added to the quarterly community meetings. we have so many team changes that i don't always keep up :)

**Review by thomas-neitmann — APPROVED** [review]

**Pull request merged**
